### PR TITLE
First version

### PR DESCRIPTION
**Pull request by evjanne**
First implementation of a GitHub action that can create and resolve Jira issues

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+default_stages: [commit]
+repos:
+  - repo: https://github.com/alessandrojcm/commitlint-pre-commit-hook
+    rev: v2.2.2
```
Comment (liimataj) [inline .pre-commit-config.yaml:4]: 4.0 seems to be the latest one.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    hooks:
+      - id: commitlint
+        stages: [commit-msg]
+        additional_dependencies: ["@commitlint/config-conventional"]
+  - repo: https://github.com/prettier/prettier
+    rev: 2.1.2
```
Comment (liimataj) [inline .pre-commit-config.yaml:10]: 2.2.1 seems to be the latest one.

File: action.yml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    description: "Account password"
+    required: true
+  action:
+    description: "Action type (create-issue/resolve-issue) needs to be provided as a parameter or optionally from the dispatch event"
+    required: false
+  title:
+    description: "Issue title, needs to be provided as a parameter or optionally from the dispatch event"
+    required: false
+  description:
+    description: "Issue title, needs to be provided as a parameter or optionally from the dispatch event"
```
Comment (liimataj) [inline action.yml:23]: issue description actually?

**Review by liimataj — APPROVED** [review]

File: .pre-commit-config.yaml (hunk 1 above)
Comment (evjanne) [reply .pre-commit-config.yaml:4]: There seems to be an open issue with version 4 so I'll go with v.3.0.0

**Review by evjuhana — COMMENTED** [review]
Some small comments

File: src/create-issue.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  console.log(body);
+  return body;
+}
+
+async function getUserLink(user) {
+  const userData = await getUser(user.login);
+  if (userData.email) {
+    return `[~${userData.email}]`;
+  }
+  return `[${userData.name || userData.login}|${userData.html_url}]`;
```
[the remark this answers is not in the document]
Comment (evjanne) [reply src/create-issue.js:65]: No, because Jira links look like `[text|url]`

File: src/gh.js [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+  const pull_number = pr.number;
+  try {
+    const reviews = await octokit.pulls.listReviews({
+      owner,
+      repo,
+      pull_number,
+    });
+    return reviews.data;
+  } catch (error) {
+    console.warn("WARNING: No reviews found");
```
[the remark this answers is not in the document]
Comment (evjanne) [reply src/gh.js]: No it should not fail that way. I added some error handling and let's see if this actually ever breaks.

File: src/gh.js [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+exports.getAuthor = async function () {
+  const pr = await exports.getPR();
+  return pr.data.user;
+};
+
+exports.getUser = async function (username) {
+  const token = core.getInput("github_token", { required: true });
+  const octokit = getOctokit(token);
+  const user = await octokit.users.getByUsername({ username });
+  return user.data;
```
[the remark this answers is not in the document]
Comment (evjanne) [reply src/gh.js]: I added small error handling, probably still fails the same way

File: src/create-issue.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (evjuhana) [reply src/create-issue.js:65]: Oh my mistake, I read the pipe to be inside the braces...

**Review by evjuhana — APPROVED** [review]

**Pull request merged**